### ISSUE TITLE
JS TypeError on opening or editing a thread conversation from a channel

### ISSUE THREAD

**Issue by canni-42** [mattermost/mattermost#26280]
#### Summary
JS TypeError on opening or editing a thread conversation from a (private) channel, causing the screen the blank out white (tab: channel); only the top navigation bar is still available. Restart of mattermost required.

#### Steps to reproduce
Environement
- Mattermost desktop client: v.5.6.0 (commit aebafea), Windows 11
- Mattermost server version: v. 9.2.4 (Database scheme 113, Build-Number 7222025858), Databse: postgresql
- Build Hashwert: 958c7a2f05ee946af7540396c911d05c72f33538, EE Build Hashwert: 9194ae642e0b263d9fef4f0f7eb1106469659dff, Build-Datum: Fri Dec 15 12:36:04 UTC 2023

Steps
- (Private) Channel
- Threaded conversation
- Edit/Opening an own message of that thread from within the channel

_This behavior is not steadily reproducable. It occurs trying to open or editor an own threaded message within a channel._ 

#### Expected behavior
- Edit/Opening the message
- Channel tab screen does not blank out white

#### Observed behavior (that appears unintentional) 
- Screen blanks out white

Developer Tools - Console - Server
```
react-dom.production.min.js:216 
TypeError: Cannot read properties of undefined (reading 'type')
    at S (post_utils.ts:116:41)
    at posts.ts:376:30
    at Array.forEach (<anonymous>)
    at posts.ts:373:29
    at index.js:96:36
    at helpers.ts:18:36
    at index.js:113:39
    at helpers.ts:18:36
    at index.js:109:45
    at index.js:53:31

[Util] handleError::  Uncaught TypeError: Cannot read properties of undefined (reading 'type')

zoom_249d6071dc514b5a_bundle.js:2 
[Util] handleError::  Uncaught TypeError: Cannot read properties of undefined (reading 'type')
E @ zoom_249d6071dc514b5a_bundle.js:2
L @ zoom_249d6071dc514b5a_bundle.js:2
(anonymous) @ zoom_249d6071dc514b5a_bundle.js:2

[Util] handleError::  Uncaught TypeError: Cannot read properties of undefined (reading 'type')

react-dom.production.min.js:123 
Uncaught TypeError: Cannot read properties of undefined (reading 'type')
    at S (post_utils.ts:116:41)
    at posts.ts:376:30
    at Array.forEach (<anonymous>)
    at posts.ts:373:29
    at index.js:96:36
    at helpers.ts:18:36
    at index.js:113:39
    at helpers.ts:18:36
    at index.js:109:45
    at index.js:53:31
```
Diagnostics
- ran after error occured
- note: the listed "uncaught exceptions" in diagnostics (step 10) does not belong to this issue
[diagnostics_22-1-2024_8-55-21-485.txt](https://github.com/mattermost/mattermost/files/14370791/diagnostics_22-1-2024_8-55-21-485.txt)

Screenshot after error occurs:
![screenshot](https://github.com/mattermost/mattermost/assets/65530234/fd677df9-2539-4ead-a988-f4d4175ded82)

**Comment by amyblais** [comment]
We fixed a bug related to this in v9.4.2 and v9.5.0 releases https://mattermost.atlassian.net/browse/MM-53149. Can you update to one of these versions to see if you can still reproduce it?

**Comment by canni-42** [comment]
> We fixed a bug related to this in v9.4.2 and v9.5.0 releases https://mattermost.atlassian.net/browse/MM-53149. Can you update to one of these versions to see if you can still reproduce it?

This seems to be the problem I am experiencing. Sorry, I did not see this when I looked at the Jira tickets.
I'll ask our IT department for an update.

**Comment by canni-42** [comment]
Update: We have updated our server version. My guess is that the bug is out, although reproducing it was a bit tricky anyway. If it reappears, I'll just reopen the ticket and update it.
